src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Internal/MsQuicConstants.cs

```diff
@@ -0,0 +1,151 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.InteropServices;
+
+namespace System.Net.Quic.Implementations.MsQuic.Internal
+{
+    internal static class MsQuicConstants
+    {
+        internal static uint InternalError = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? Windows.InternalError : Linux.InternalError;
+        internal static uint Success = 0;
+        internal static uint Pending = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? Windows.Pending : Linux.Pending;
+        private const uint SuccessConst = 0;
+
+        internal static Func<uint, string> ErrorTypeFromErrorCode = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? Windows.GetError : (Func<uint, string>)Linux.GetError;
```

Comment:
BTW: This library is setup to have different builds for Windows and Unix already: https://github.com/dotnet/runtime/blob/master/src/libraries/System.Net.Quic/src/Configurations.props#L4

Reply:
I'm not familiar with how CoreFx does OS specific dependencies. Would conditional compilation make it so when I create a System.Net.Quic.dll, there would be different dlls for different OSs?

Reply:
Yes. Take a look at the csproj for e.g. System.Net.Sockets and you'll see different files included depending on target platform.

Reply:
I'd like to make sure we are all on the same page here. I chatted with @jkotas offline about how CoreFx does interop dependencies, and it's distinctly different that what ASP.NET Core does.

In ASP.NET Core, we do our best to produce a single dll that works cross platform. For example, we do this with Libuv today https://github.com/aspnet/AspNetCore/blob/master/src/Servers/Kestrel/Transport.Libuv/src/Internal/LibuvConstants.cs. We do runtime checks when comparing constants. This distinctly contrasts what Corefx does by always splitting on different OS versions, creating separate dlls.

Reasons to have a single dll:
- Makes it so components on windows can be copied to linux and just work
Reasons to have multiple dlls:
- Save potential runtime checks for OS version
- Fits the model in corefx (which is where this project is now)

MsQuic pretty much has the same calls between OSs as well as the same signatures. The only difference is the name of the dll (msquic.dll vs libmsquic.so) and the constants returned being platform specific. So if I were to do platform specific compilation, most of it would be shared anyways.

I personally don't have much of a preference between these two options; I just want people to agree on an approach and I'll do it. @davidfowl